File: Frontend/src/types/security.ts
export interface UserSession {
  id: string;
  ip_address: string;
  user_agent: string;
  location: GeoLocation;
  browser_fingerprint: string;
  created_at: string;
  last_activity: string;
  is_suspicious: boolean;
}

export interface GeoLocation {
  country: string;
  city: string;
  latitude: number;
  longitude: number;
}

export type AlertType = 'vpn_detected' | 'suspicious_login' | 'blocked' | 'security_audit';
export type AlertSeverity = 'low' | 'medium' | 'high' | 'critical';

export interface SecurityAlert {
  id: string;
  user_id: number;
  alert_type: AlertType;
  severity: AlertSeverity;
  details: Record<string, unknown>;
  timestamp: string;
  resolved: boolean;
  resolved_by?: number;
  resolution_notes?: string;
}

export interface UserSecurityProfile {
  user_id: number;
  risk_score: number;
  known_ips: string[];
  known_devices: string[];
  known_locations: GeoLocation[];
  vpn_usage_history: VpnUsage[];
  suspicious_activities: SecurityAlert[];
  last_security_audit: string;
  security_notes?: string;
}

export interface VpnUsage {
  timestamp: string;
  ip_address: string;
  vpn_provider?: string;
  location: GeoLocation;
}

export interface AdminUser {
  id: number;
  name: string;
  email: string;
  role: 'admin' | 'super_admin';
  sessions: UserSession[];
  security_profile: UserSecurityProfile;
  last_login: string;
  is_active: boolean;
}

File: Backend/app/schemas/security.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List

class SecurityAlert(BaseModel):
    user_id: int
    alert_type: str  # suspicious_login, multiple_ips, vpn_detected, etc
    severity: str    # low, medium, high, critical
    details: dict
    timestamp: datetime
    resolved: bool = False
    resolved_by: Optional[int] = None
    resolution_notes: Optional[str] = None

class UserSecurityProfile(BaseModel):
    user_id: int
    risk_score: float
    known_ips: List[str]
    known_devices: List[str]
    known_locations: List[dict]
    vpn_usage_history: List[dict]
    suspicious_activities: List[SecurityAlert]
    last_security_audit: datetime
    security_notes: Optional[str]

class BrowserFingerprint(BaseModel):
    user_agent: str
    screen_resolution: str
    color_depth: int
    timezone: str
    plugins: List[str]
    fonts: List[str]
    canvas_hash: str
    webgl_hash: str
    audio_hash: str
    cpu_class: Optional[str]
    platform: str
    do_not_track: Optional[bool]
    hardware_concurrency: int
    device_memory: Optional[float]

class SecurityAuditLog(BaseModel):
    timestamp: datetime
    action: str
    admin_id: int
    target_user_id: Optional[int]
    details: dict
    ip_address: str
    geolocation: dict
